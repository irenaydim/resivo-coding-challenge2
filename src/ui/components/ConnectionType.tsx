import Typography from '@mui/material/Typography';
import Stack from '@mui/material/Stack';
import { ConnectionType as ConnType } from "@/models/ConnectionType";
import ElectricalServicesRoundedIcon from '@mui/icons-material/ElectricalServicesRounded';
import SensorsRoundedIcon from '@mui/icons-material/SensorsRounded';

interface ConnectionTypeProps {
  type: ConnType;
}

const ConnectionType = ({ type }: ConnectionTypeProps) => {
  return <Stack direction='row' alignItems='center' gap={0.5}>
    {type === 'wired' ? <ElectricalServicesRoundedIcon fontSize="small" /> : <SensorsRoundedIcon fontSize="small" />}
    <Typography>{type}</Typography>
  </Stack>
}

export default ConnectionType
