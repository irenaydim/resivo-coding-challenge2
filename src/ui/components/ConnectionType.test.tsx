import { render } from '@testing-library/react';
import { ConnectionType as ConnType } from '@/models/ConnectionType';
import ConnectionType from './ConnectionType';

const connectionType: ConnType = 'wired';

describe('ConnectionType', () => {
  it('should render correctly', () => {
    const { container } = render(<ConnectionType type={connectionType} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
