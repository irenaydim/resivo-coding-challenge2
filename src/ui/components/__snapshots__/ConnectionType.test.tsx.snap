// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatus should render correctly 1`] = `
<div
  class="css-1d9cypr-MuiStack-root"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeSmall css-1efkf0c-MuiSvgIcon-root"
    data-testid="PowerOffRoundedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M18 13.66V8.99c0-1-1.01-2-2-1.99V4c0-.55-.45-1-1-1s-1 .45-1 1v3h-3.88l7.63 7.63c.15-.3.25-.63.25-.97zM10 4c0-.55-.45-1-1-1s-1 .45-1 1v.88l2 2V4zm10.15 15.86-7.66-7.66-5.1-5.1-2.56-2.56a.9959.9959 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l2.63 2.63c-.03.13-.05.27-.05.41v4.66c0 .53.21 1.04.58 1.41L9.5 18v2c0 .55.45 1 1 1h3c.55 0 1-.45 1-1v-2l.48-.48 3.76 3.76c.39.39 1.02.39 1.41 0 .39-.39.39-1.03 0-1.42z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1nkhxq8-MuiTypography-root"
  >
    offline
  </p>
</div>
`;

exports[`ConnectionType should render correctly 1`] = `
<div
  class="css-1jtdgec-MuiStack-root"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
    data-testid="ElectricalServicesRoundedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M21 14c0-.55-.45-1-1-1h-2v2h2c.55 0 1-.45 1-1zm-1 3h-2v2h2c.55 0 1-.45 1-1s-.45-1-1-1zm-4-5h-2c-1.1 0-2 .9-2 2h-1c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1h1c0 1.1.9 2 2 2h2c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zM5 13c0-1.1.9-2 2-2h1.5c1.93 0 3.5-1.57 3.5-3.5S10.43 4 8.5 4H5c-.55 0-1 .45-1 1s.45 1 1 1h3.5c.83 0 1.5.67 1.5 1.5S9.33 9 8.5 9H7c-2.21 0-4 1.79-4 4s1.79 4 4 4h2v-2H7c-1.1 0-2-.9-2-2z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    wired
  </p>
</div>
`;
